WcaOnRails/spec/features/delegate_report_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.feature "Registration management" do
+  let(:delegate) { FactoryGirl.create :delegate, name: "Jeremy on Bart" }
+  let(:competition) { FactoryGirl.create :competition, :registration_open, delegates: [delegate], name: "Submit Report 2017" }
+
+  context "when signed in as competition delegate" do
+    before :each do
+      sign_in delegate
+    end
+
+    scenario "view, edit, save, submit report" do
+      # View report
+      visit "/competitions/#{competition.id}/report"
+      expect(page).to have_text("Submit Report 2017")
```

Comment:
Ah, yeah, I prefer to be wetter in tests. The two examples you gave for this test are pretty simple to compute at runtime as you did, but at some point, your test code ends up looking like the very code you're trying to test, and my brain starts to get stuck in "what am I even doing" territory, which is a dark place I try to avoid.